embassy-stm32/src/sdmmc/mod.rs

```diff
@@ -1032,12 +1032,11 @@ impl<'d, T: Instance> Sdmmc<'d, T> {
 
     /// Wait for a previously started datapath transfer to complete from an interrupt.
     #[inline]
+    #[allow(unused)]
     async fn complete_datapath_transfer(block: bool) -> Result<(), Error> {
-        let regs = T::regs();
-
         let res = poll_fn(|cx| {
             T::state().register(cx.waker());
-            let status = regs.star().read();
+            let status = T::regs().star().read();
```

Comment:
If you were to revert this, and then put `compiler_fence(Ordering::SeqCst)` after registering the waker, is the problem resolved?

Reply:
For what its worth, searching around the codebase, most `poll_fn` usage calls `T::regs()` within the wrapped function, rather than using a previously stored variable.